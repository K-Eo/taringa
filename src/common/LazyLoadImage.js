import React from 'react'
import LazyLoad from 'react-lazyload'

import './LazyLoadImage.css'
import Image from './Image'

const LazyLoadImage = ({ url, ...rest }) => {
  return (
    <Image {...rest}>
      <LazyLoad
        debounce
        once
        placeholder={<div className="LazyLoad-placeholder" />}
      >
        <img src={url} alt={url} className="Image-source rounded" />
      </LazyLoad>
    </Image>
  )
}

LazyLoadImage.defaultProps = {
  url: '#IMAGE_URL',
}

export default LazyLoadImage
