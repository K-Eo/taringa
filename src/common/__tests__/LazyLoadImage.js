import React from 'react'

import LazyLoadImage from '../LazyLoadImage'

describe('LazyLoadImage', () => {
  it('exists', () => {
    expect(LazyLoadImage).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<LazyLoadImage />)

    expect(wrapper).toMatchSnapshot()
  })
})
