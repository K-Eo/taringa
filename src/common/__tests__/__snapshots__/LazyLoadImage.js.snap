// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyLoadImage renders default 1`] = `
<Image
  height={1}
  url="#IMAGE_URL"
  width={1}
>
  <LazyLoad
    debounce={true}
    offset={0}
    once={true}
    overflow={false}
    placeholder={
      <div
        className="LazyLoad-placeholder"
      />
    }
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt="#IMAGE_URL"
      className="Image-source rounded"
      src="#IMAGE_URL"
    />
  </LazyLoad>
</Image>
`;
